libtcmu_config.c

```diff
@@ -149,11 +151,19 @@ static void tcmu_conf_set_options(struct tcmu_config *cfg, bool reloading)
 		 * 4, default /var/log/
 		 */
 		if (!tcmu_get_logdir())
-			tcmu_logdir_create(cfg->log_dir_path);
+			tcmu_logdir_create(cfg->log_dir_path, false);
 		else
 			tcmu_warn("The logdir option from the tcmu.conf will be ignored\n");
 	} else {
-		tcmu_warn("The logdir option is not supported by dynamic reloading for now!\n");
+		/*
+		 * Here we asume that users want to change the
+		 * log_dir_path without considering tha priority
+		 * mentioned above.
+		 */
+		tcmu_logdir_create(cfg->log_dir_path, true);
+		ret = tcmu_create_file_output(tcmu_get_log_level(), TCMU_LOG_FILENAME, true);
```

Comment:
80 char cols.

Reply:
I think a general issue here is that we do

tcmu_destroy_log()
then
tcmu_destroy_config()

so if during the log resources are freed and destroyed then someone does a update, this would leak and access freed resources.

I think we need to change the ordering or add some new log function that handles checking if the log has been torn down before trying to update it.

Reply:
Hey, did you handle this last comment? We need to change the shutdown ordering, or do something so if logging is shutdown while the config update is going on, we do not race?